extension.ts

```diff
@@ -103,10 +103,27 @@ function initializeExtensionVariables(ctx: vscode.ExtensionContext): void {
   registerUIExtensionVariables(ext);
 }
 
+async function setContextKeys(ctx: vscode.ExtensionContext): Promise<void> {
+  const isRemote = ctx.executionContext === vscode.ExtensionExecutionContext.Remote;
+  vscode.commands.executeCommand('setContext', 'docker:isRemote', isRemote);
+
+  const hasRemote = vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders[0].uri.authority;
+  const hasShownInfoMessage = ctx.globalState.get('docker.hasShownRemoteNotification');
+  if (!hasShownInfoMessage && !isRemote && hasRemote) {
```

Comment:
So we're worried it'll be confusing if commands are just hidden and I'm not sure how effective a single global notification would be. Instead, what if we have a per-command warning like this:
```
This command requires access to the remote workspace, but the Docker extension is running on your local machine.
```
In this case, I think it also makes sense to have a button that would switch Docker from a ui extension to a workspace extension, since that seems difficult to figure out.